nifi-framework-bundle/nifi-framework-extensions/nifi-framework-kubernetes-bundle/nifi-framework-kubernetes-state-provider/src/main/java/org/apache/nifi/kubernetes/state/provider/KubernetesConfigMapStateProvider.java

```diff
@@ -272,7 +237,12 @@ public StateMap getState(final String componentId) throws IOException {
      */
     @Override
     public boolean replace(final StateMap currentState, final Map<String, String> state, final String componentId) throws IOException {
-        final ConfigMapBuilder configMapBuilder = createConfigMapBuilder(state, componentId);
+        if (!(currentState instanceof StandardStateMap)) {
+            throw new IllegalStateException("Current state is not an instance of StandardStateMap");
+        }
+
+        final Optional<ObjectMeta> existingMetadata = ((StandardStateMap) currentState).getConfigMapMetadata();
```

Comment:
The casting is a bit verbose as written. Instead, what do you think about using pattern variables as `if (currentState instanceof StandardStateMap standardStateMap) { ... } else { throw new IllegalStateException() }`